core/primitives/src/validator_mandates.rs

```diff
@@ -125,35 +131,40 @@ impl ValidatorMandates {
         // assigned the mandates with indices 1, 5, and 9.
         //
         // TODO(#10014) shuffle shard ids to avoid a bias towards smaller shard ids
-        let mut mandates_per_shard = Vec::with_capacity(self.config.num_shards);
+        let mut mandates_per_shard: ValidatorMandatesAssignment =
+            vec![HashMap::new(); self.config.num_shards];
         for shard_id in 0..self.config.num_shards {
```

Comment:
The name `shard_id` helps to point out what is interated. I think it’s not problematic since mandates are actually collected for `shard_id` and then written to position `*.get_alias(shard_id)`. With an abstract `i` that might be harder to follow since there’s a lot going on here.